packages/kbn-securitysolution-list-utils/src/helpers/index.ts

```diff
@@ -291,57 +286,6 @@ export const getUpdatedEntriesOnDelete = (
   }
 };
 
-/**
- * Returns filtered index patterns based on the field - if a user selects to
- * add nested entry, should only show nested fields, if item is the parent
- * field of a nested entry, we only display the parent field
- *
- * @param patterns DataViewBase containing available fields on rule index
- * @param item exception item entry
- * set to add a nested field
- */
-export const getFilteredIndexPatterns = (
-  patterns: DataViewBase,
-  item: FormattedBuilderEntry,
-  type: ExceptionListType,
-  preFilter?: (i: DataViewBase, t: ExceptionListType, o?: OsTypeArray) => DataViewBase,
-  osTypes?: OsTypeArray
-): DataViewBase => {
-  const indexPatterns = preFilter != null ? preFilter(patterns, type, osTypes) : patterns;
-
-  if (item.nested === 'child' && item.parent != null) {
-    // when user has selected a nested entry, only fields with the common parent are shown
-    return {
-      ...indexPatterns,
-      fields: indexPatterns.fields
-        .filter((indexField) => {
-          const subTypeNested = getDataViewFieldSubtypeNested(indexField);
-          const fieldHasCommonParentPath =
-            subTypeNested &&
-            item.parent != null &&
-            subTypeNested.nested.path === item.parent.parent.field;
-
-          return fieldHasCommonParentPath;
-        })
-        .map((f) => {
-          const [fieldNameWithoutParentPath] = f.name.split('.').slice(-1);
-          return { ...f, name: fieldNameWithoutParentPath };
-        }),
-    };
-  } else if (item.nested === 'parent' && item.field != null) {
-    // when user has selected a nested entry, right above it we show the common parent
-    return { ...indexPatterns, fields: [item.field] };
-  } else if (item.nested === 'parent' && item.field == null) {
-    // when user selects to add a nested entry, only nested fields are shown as options
-    return {
-      ...indexPatterns,
-      fields: indexPatterns.fields.filter((field) => isDataViewFieldSubtypeNested(field)),
-    };
-  } else {
-    return indexPatterns;
-  }
-};
-
```

Comment:
I needed to move this function out of the package because packages can't utilize types from plugins. Specifically, I needed the `DataView` type generated by the `data-views-plugin` `DataView` class to pass into the `preFilter` param.